apicast/src/backend_client.lua

```diff
@@ -94,15 +89,30 @@ local function call_backend_transaction(self, path, ...)
     end
   end
 
-  local url = resty_url.join(endpoint, '/transactions', path .. '?' .. concat(query, '&'))
+  return resty_url.join(endpoint, '', path .. '?' .. concat(query, '&'))
+end
+
+local function call_backend_transaction(self, path, options, ...)
+  local http_client = self.http_client
+
+  if not http_client then
+    return nil, 'not initialized'
+  end
 
-  local res = http_client.get(url)
+  local url = build_url(self, path, ...)
+  local res = http_client.get(url, options)
 
   ngx.log(ngx.INFO, 'backend client uri: ', url, ' ok: ', res.ok, ' status: ', res.status, ' body: ', res.body)
 
   return res
 end
 
+local authorize_options = {
+  headers = {
+    ['3scale-options'] =  'rejection_reason_header=1'
```

Comment:
@davidor this should be more dynamic, right? So we can have `no_body` for normal authrep but not for oauth calls.

Reply:
Yes, we need that.

Reply:
Let's leave this for a future PR when we actually add more options to avoid adding more complexity to this PR.